src/Microsoft.OData.Core/UriParser/Resolver/ODataUriResolver.cs

```diff
@@ -545,11 +579,11 @@ internal static ODataUriResolver GetUriResolver(IServiceProvider container)
             }
         }
 
-        private static NormalizedSchemaElementsCache GetNormalizedSchemaElementsCache(IEdmModel model)
+        private static NormalizedModelElementsCache GetNormalizedModelElementsCache(IEdmModel model)
         {
             Debug.Assert(model != null);

```

Comment:
why are doing these checks here if you already did them in the constructor?

Reply:
I could remove it. I think someone else suggested having it here just to be sure. But I don't think it makes a major difference to have it here or not.

Reply:
I removed it. Added argument null checks to public methods in this class instead.